File: julia/ch1/Ex113.jl
"""
# module Ex113

- Julia version: 0.6.4
- Author: Yohan
- Date: 2019-02-03

# Examples

```jldoctest
julia> include("c:/Users/Yohan/IdeaProjects/sicp/julia/ch1/ex112.jl")

julia> Ex113.fib(15) #610
```
"""
module Ex113

psi(n) = ((1.0 + sqrt(5)) / 2) ^ n

phi(n) = ((1.0 - sqrt(5)) / 2) ^ n

fib(n) = (n < 2) ? n : round(Int64, (psi(n) - phi(n)) / sqrt(5))

end

File: octave/ch1/Ex113.m
function Ex113
  fib(0)
  fib(1)
  fib(2)
  fib(3)
  fib(10)
  fib(15)
endfunction

function retval = fib(n)
  if n < 2
    retval = n;
    return;
  else
    retval = round((psi(n) - phi(n)) / sqrt(5));
  endif
endfunctionpp

function retval = psi(n)
  retval = ((1 + sqrt(5)) / 2) ^ n;
  return; 
endfunction

function retval = phi(n)
  retval = ((1 - sqrt(5)) / 2) ^ n;
  return; 
endfunction
